filter_plugins/filters.py

```diff
@@ -136,7 +136,10 @@ def ether_check(context, interface):
     :param interface: An item in interfaces_ether_interfaces.
     :returns: A dict containing 'diff' and 'reason' items.
     """
-    result = _interface_check(context, interface, "ether")
+    if interface['type'] == 'ipoib':
```

Comment:
I can see the appeal of this approach. We should make the type field optional for backwards compatibility:

```
if interface.get('type') == 'ipoib':
```

VLANs won't work for IB, but otherwise I guess it should be the same.

Reply:
I have updated the conditional as you suggested so the type field is optional